artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/file/JDBCSequentialFileFactory.java

```diff
@@ -0,0 +1,255 @@
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+import org.apache.activemq.artemis.core.io.SequentialFile;
+import org.apache.activemq.artemis.core.io.SequentialFileFactory;
+import org.apache.activemq.artemis.core.server.ActiveMQComponent;
+import org.apache.activemq.artemis.jdbc.store.JDBCUtils;
+import org.apache.activemq.artemis.jdbc.store.file.sql.SQLProvider;
+import org.apache.activemq.artemis.journal.ActiveMQJournalLogger;
+
+public class JDBCSequentialFileFactory implements SequentialFileFactory, ActiveMQComponent {
+
+   private Connection connection;
+
+   private String connectionUrl;
+
+   private final Driver driver;
+
+   private boolean started;
+
+   private final String tableName;
+
+   private List<JDBCSequentialFile> files;
+
+   private PreparedStatement selectFileNamesByExtension;
+
+   private Executor executor;
+
+   private SQLProvider sqlProvider;
+
+   private Map<String, Object> fileLocks = new HashMap<>();
+
+   public JDBCSequentialFileFactory(final String connectionUrl,
+                                    final String tableName,
+                                    final String className,
+                                    Executor executor) throws Exception {
+      this.connectionUrl = connectionUrl;
+      this.executor = executor;
+      this.tableName = tableName.toUpperCase();
+
+      files = new ArrayList<>();
+      sqlProvider = JDBCUtils.getSQLProvider(JDBCUtils.getDriver(className).getClass().getCanonicalName(), tableName);
+      driver = JDBCUtils.getDriver(className);
+   }
+
+   public Connection getConnection() {
+      return connection;
+   }
+
+   @Override
+   public SequentialFile createSequentialFile(String fileName) {
+      try {
+         fileLocks.putIfAbsent(fileName, new Object());
+         JDBCSequentialFile file = new JDBCSequentialFile(this, fileName, sqlProvider, executor, fileLocks.get(fileName));
+         files.add(file);
+         return file;
+      }
+      catch (Exception e) {
+         ActiveMQJournalLogger.LOGGER.error("Could not create file", e);
+      }
+      return null;
+   }
+
+   @Override
+   public int getMaxIO() {
+      return 1;
+   }
+
+   @Override
+   public List<String> listFiles(String extension) throws Exception {
+      List<String> fileNames = new ArrayList<>();
+
+      selectFileNamesByExtension.setString(1, extension);
+      try (ResultSet rs = selectFileNamesByExtension.executeQuery()) {
+         while (rs.next()) {
+            fileNames.add(rs.getString(1));
+         }
+      }
+      return fileNames;
+   }
+
+   @Override
+   public boolean isSupportsCallbacks() {
+      return true;
+   }
+
+   @Override
+   public void onIOError(Exception exception, String message, SequentialFile file) {
+
+   }
+
+   // Methods lifted from NIO Impl
+   @Override
+   public ByteBuffer allocateDirectBuffer(final int size) {
+      // Using direct buffer, as described on https://jira.jboss.org/browse/HORNETQ-467
```

Comment:
Copy and Paste from the NIOSequentialFileFactory.  Actually this is a mistake I thought I had refactored this into a static method on the NIOSequentialFileFactory class and called it here.  I will update.  Thanks.